samples/csharp_dotnetcore/17.multilingual-bot/Startup.cs

```diff
@@ -26,168 +19,54 @@ namespace Microsoft.BotBuilderSamples
-        /// </summary>
-        /// <param name="services">Specifies the contract for a <see cref="IServiceCollection"/> of service descriptors.</param>
+        // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddBot<MultiLingualBot>(options =>
-            {
-                var secretKey = Configuration.GetSection("botFileSecret")?.Value;
-                var botFilePath = Configuration.GetSection("botFilePath")?.Value;
-                if (!File.Exists(botFilePath))
-                {
-                    throw new FileNotFoundException($"The .bot configuration file was not found. botFilePath: {botFilePath}");
-                }
-
-                // Loads .bot configuration file and adds a singleton that your Bot can access through dependency injection.
-                var botConfig = BotConfiguration.Load(botFilePath ?? @".\multilingual-bot.bot", secretKey);
-                services.AddSingleton(sp => botConfig ?? throw new InvalidOperationException($"The .bot configuration file could not be loaded. botFilePath: {botFilePath}"));
-              
-                // Retrieve current endpoint.
-                var environment = _isProduction ? "production" : "development";
-                var service = botConfig.Services.FirstOrDefault(s => s.Type == "endpoint" && s.Name == environment);
-                if (!(service is EndpointService endpointService))
-                {
-                    throw new InvalidOperationException($"The .bot file does not contain an endpoint with name '{environment}'.");
-                }
-
-                options.CredentialProvider = new SimpleCredentialProvider(endpointService.AppId, endpointService.AppPassword);
-
-                // Creates a logger for the application to use.
-                ILogger logger = _loggerFactory.CreateLogger<MultiLingualBot>();
-
-                // Catches any errors that occur during a conversation turn and logs them.
-                options.OnTurnError = async (turnContext, exception) =>
-                {
-                    logger.LogError($"Exception caught : {exception}");
+            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
 
-                    // By-pass the middleware by sending the Activity directly on the Adapter.
-                    var activity = MessageFactory.Text("Sorry, it looks like something went wrong.");
-                    activity.ApplyConversationReference(turnContext.Activity.GetConversationReference());
-                    await turnContext.Adapter.SendActivitiesAsync(turnContext, new[] { activity }, default(CancellationToken));
-                };
+            // Create the credential provider to be used with the Bot Framework Adapter.
+            services.AddSingleton<ICredentialProvider, ConfigurationCredentialProvider>();
```

Comment:
John stopped by yesterday, and mentioned there is a ConfigurationCredentialProvider in the integration library.  It makes sense to use it, rather than create one in each sample: https://github.com/Microsoft/botbuilder-dotnet/blob/master/libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/ConfigurationCredentialProvider.cs